src/transformers/pipelines/fill_mask.py

```diff
@@ -115,25 +115,56 @@ def __call__(self, *args, targets=None, top_k: Optional[int] = None, **kwargs):
         inputs = self._parse_and_tokenize(*args, **kwargs)
         outputs = self._forward(inputs, return_tensors=True)
 
+        # top_k must be defined
+        if top_k is None:
+            top_k = self.top_k
+
         results = []
         batch_size = outputs.shape[0] if self.framework == "tf" else outputs.size(0)
 
         if targets is not None:
-            if len(targets) == 0 or len(targets[0]) == 0:
-                raise ValueError("At least one target must be provided when passed.")
             if isinstance(targets, str):
                 targets = [targets]
 
-            targets_proc = []
+            try:
+                vocab = self.tokenizer.get_vocab()
+            except Exception:
```

Comment:
Do you think we could have a better exception than this?

Reply:
Depends if we want to be permissive or restrictive in the type of error we catch.
Ideally there would not be a try except. But I was not sure it was mandatory to have such a method.

If we consider this could fail for valid reasons (not implemented), I don't really think we should make assumptions about the type of exceptions. NotImplementedError, AttributeError, but it could very well be a signature issue or something else.

I think we should either keep it like this (very permissive in types of errors) or remove the try altogether and be very restrictive (tokenizers MUST implement a valid `get_vocab`)

Reply:
Alright, works for me!